core/filter.js

```diff
@@ -2242,7 +2242,7 @@
 				return;
 			}
 
-			var widths = element.styles.margin.match( /(\-?[\.\d]+\w+)/g ) || [ '0px' ];
+			var widths = element.styles.margin.match( /(\-?[\.\d]*\w+)/g ) || [ '0px' ];
```

Comment:
Changes in this regex made it work with `0`/`auto` values and it works better than before indeed. However, analyzing it deeper I'm not sure if it shouldn't be changed completely :thinking: 

The previous version also accepts invalid values (e.g. two or more digit numbers without units):

![](https://user-images.githubusercontent.com/1061942/93349350-a5657180-f837-11ea-909c-2110f6bcd366.png)

And yours while fixing initial issue also accepts invalid values (any digit without units):

![image](https://user-images.githubusercontent.com/1061942/93349450-ca59e480-f837-11ea-9d97-3d1f78b4fd3a.png)

I'm not sure how far we should polish it TBH because it may get complex. CSS supports percentage here too, which our regex can't handle (well it can but results in extracting number only which is not correct handling)...

But ok, let's be reasonable here - this function job is to simply split margin shorthand and not to validate what's there. The browser will simply not render the margin if values are incorrect. So this means all valid values should be used in the same form in split properties. And the ones we handle incorrectly are percentage values (because e.g. `10%` will be translated to `10` which is incorrect) and so lets try to additionally handle something like (please also add unit test for the `%` case):

```
5%
auto 10%
123px -10.5% 11.5
0% 10em 12.3px 1234%
```

---

I also wonder about `|| [ '0px' ]` part. Since we check if styles are empty in the beginning, the regex should always return some result, unless... and we are back to introduced change in regex. Previous one didn't match text values:
![image](https://user-images.githubusercontent.com/1061942/93351483-1a39ab00-f83a-11ea-9501-0fbf2fe30646.png)

while this one does:

![image](https://user-images.githubusercontent.com/1061942/93351543-2887c700-f83a-11ea-89ac-da440e44520d.png)

---

To sum up, I would be for modifying the regex in a way to not accepted many more than required - so `0` and `auto`, which are in fact special cases and the rest are just numbers + units. And so the regex could be extended like:

```
(auto|0|(?:\-?[\.\d]+\w+))
```
It will match `0` (unit doesn't matter here so will be skipped), match `auto` or match number + unit. Or more precisely:

```
(auto|0|(?:\-?[\.\d]+(?:\w+|%)))
``` 
to match `%` too (if you see it can be improved or made shorter feel free to change it).

Still it kind of validates input not allowing single digits without units, but allowing e.g `12` (due to `\w+` which also includes digits), not allowing text only, etc.  But let's keep it like that, fixing the way `0`, `auto` and `%` are handled as mentioned above.